Helper/Order.php

```diff
@@ -2422,18 +2422,27 @@ public function createOrderInvoice($order, $amount, $notify = false, $transactio
         $invoice->setRequestedCaptureCase(Invoice::CAPTURE_OFFLINE);
         $invoice->setTransactionId($transactionId);
         $invoice->setGrandTotal($amount);
+        $invoice->setBaseGrandTotal($amount);
         $invoice->register();
         $invoice->save();
 
         $order->addRelatedObject($invoice);
 
+        // pre-save required order data with will be overwritten during invoice email sending
+        $baseSubtotal = $order->getBaseSubtotal();
+        $baseTaxAmount = $order->getBaseTaxAmount();
+        $baseShippingAmount = $order->getBaseShippingAmount();
         if ($this->scopeConfig->isSetFlag(
             InvoiceEmailIdentity::XML_PATH_EMAIL_ENABLED,
             ScopeInterface::SCOPE_STORE,
             $order->getStoreId()
         ) && !$invoice->getEmailSent()) {
             try {
                 $this->invoiceSender->send($invoice);
+                // restore order data
+                $order->setBaseSubtotal($baseSubtotal);
```

Comment:
How about restoring data after try{} catch block to do it even if it was the exception for some reason?

Reply:
Good catch, thanks, updated.